quodlibet/ext/events/musicbrainzsync.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2019 LoveIsGrief
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+from quodlibet import plugins, qltk
+from quodlibet.qltk.entry import UndoEntry
+from quodlibet.util.thread import call_async_background, Cancellable
+
+try:
+    import musicbrainzngs
+except ImportError:
+    raise plugins.MissingModulePluginException("musicbrainzngs")
+
+from gi.repository import Gtk
+from quodlibet import _
+from quodlibet.plugins import PluginConfig
+from quodlibet.plugins.events import EventPlugin
+
+ATTR_BRAINZ = 'musicbrainz_trackid'
+ATTR_RATING = '~#rating'
+
+BRAINZ_APP = "quodlibetMusicBrainzSync"
+VERSION = "0.1"
+
+plugin_config = PluginConfig("musicbrainz-sync")
+defaults = plugin_config.defaults
+defaults.set("username", "")
+defaults.set("password", "")
+
+
+class MusicBrainzSyncPlugin(EventPlugin):
+    PLUGIN_ID = "musicbrainzsync"
+    VERSION = VERSION
+    PLUGIN_NAME = _("MusicBrainz Sync")
+    PLUGIN_DESC = _("Syncs the rating of a song with music brainz.")
+
+    def __init__(self):
+        super(MusicBrainzSyncPlugin, self).__init__()
+        musicbrainzngs.set_rate_limit()
+        musicbrainzngs.set_useragent(
+            BRAINZ_APP, VERSION, "loveisgrief@tuta.io"
+        )
+        musicbrainzngs.auth(
+            plugin_config.get("username"),
+            plugin_config.get("password")
+        )
+
+    def plugin_on_changed(self, songs):
+        ratings_dict = {
+            song[ATTR_BRAINZ]: int(song[ATTR_RATING] * 100)
+            for song in songs
+            if ATTR_BRAINZ in song and ATTR_RATING in song
+        }
+        if len(ratings_dict) > 0:
+            call_async_background(
+                musicbrainzngs.submit_ratings,
+                Cancellable(),
+                callback=lambda *args: (),
```

Comment:
I guess `None` doesn't work here?

Reply:
Unfortunately not because of https://github.com/quodlibet/quodlibet/blob/4b8a99f0dde0ee87d38739edb228d37ce93f8665/quodlibet/util/thread.py#L104